mycpp/gc_heap.h

```diff
@@ -957,23 +957,21 @@ class List : public gc_heap::Obj {
     auto self = this;
     StackRoots _roots({&self});
 
-    if (self->capacity_ < n) {
-      // Example: The user asks for space for 7 integers.  Account for the
-      // header, and say we need 9 to determine the obj length.  9 is
-      // rounded up to 16, for a 64-byte obj.  Then we actually have space
-      // for 14 items.
-      self->capacity_ = RoundUp(n + kCapacityAdjust) - kCapacityAdjust;
-      auto new_slab = NewSlab<T>(self->capacity_);
-
-      // slab_ may not be initialized constructor because many lists are
-      // empty.
-      if (self->capacity_ != 0) {
-        // log("Copying %d bytes", len_ * sizeof(T));
-        memcpy(new_slab->items_, self->slab_->items_, self->len_ * sizeof(T));
-      }
-      self->slab_ = new_slab;
+    // Don't do anything if there's already enough space.
+    if (self->capacity_ >= n) return;
```

Comment:
Note: as a micro-optimisation, this could be moved up to the top of this function. The question is: do you you want to follow a simple/conservative rule like: "member functions always start with `auto self = this; ..`". Or do you want it to be case-by-case?

Reply:
True, I'm not sure ... I think this is nice for now, to keep the code looking regular.  But the style may evolve as we make it correct!